ddl/partition.go

```diff
@@ -3572,13 +3580,17 @@ func buildCheckSQLConditionForRangeExprPartition(pi *model.PartitionInfo, index
 		paramList = append(paramList, driver.UnwrapFromSingleQuotes(pi.Definitions[index].LessThan[0]))
 	} else if index == len(pi.Definitions)-1 && strings.EqualFold(pi.Definitions[index].LessThan[0], partitionMaxValue) {
 		buf.WriteString(pi.Expr)
-		buf.WriteString(" < %?")
+		buf.WriteString(" < %? or ")
+		buf.WriteString(pi.Expr)
+		buf.WriteString(" is null")
```

Comment:
Can there be a case where the expression string also would need to be wrapped into parentheses?

Reply:
I have to say i'm not sure about it. I tested `partition by range (id+1)`, the expression is already wapped. And if `pi.Expr is null` needed, then `pi.Expr < %?` seems also needs. I think we can keep it now before we can make sure some cases as you described exist.